addons/medical_damage/functions/fnc_applyDamage.sqf

```diff
@@ -0,0 +1,220 @@
+#include "script_component.hpp"
+/*
+ * Author: gandjustas
+ * Converts hitpoint damages to body part damages, checks heart hit.
+ *
+ * Arguments:
+ * 0: Unit That Was Hit <OBJECT>
+ * 1: Source of damage <OBJECT>
+ * 2: Ammo <STRING>
+ * 3: Structural damage <NUMBER>
+ * 4: Array of damage to hitpoints <ARRAY>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [player, objNull, "fire", 0.1] call ace_medical_damage_fnc_applyDamage
+ *
+ * Public: No
+ */
+
+params ["_unit", "_source", "_ammo", "_damage", ["_hitpointDamages",[]]];
+
+if (alive _unit) then { _unit setDamage 0; };
+
+private _typeOfDamage = _ammo call FUNC(getTypeOfDamage);
+TRACE_6("Received hit",_unit,_source,_ammo,_typeOfDamage,_damage,_hitpointDamages);
+
+// No damage to hitpoints - special cases or environmental damage
+if (_hitpointDamages isEqualTo []) then {
+    switch (_typeOfDamage) do
+    {
+        case "fire": {
+            TRACE_4("Burning",_unit,_source,_instigator,_damage);        
+
+            private _damageSelectionArray = [HITPOINT_INDEX_BODY, 1, HITPOINT_INDEX_LLEG, 1, HITPOINT_INDEX_RLEG, 1];
+            private _damagedBodyPart = selectRandomWeighted _damageSelectionArray;
+
+            [QEGVAR(medical,woundReceived), [
+                _unit, 
+                _damagedBodyPart, 
+                _damage, 
+                "unknown", 
+                [[_damagedBodyPart, _damage / (GVAR(structuralDamageToBodyPartCoeff) select _damagedBodyPart)]]
+                ]
+            ] call CBA_fnc_localEvent;
+        };
+        case "drowning": {
+            TRACE_5("Drowning",_unit,_source,_instigator,_damage,_newDamage);
+            [QEGVAR(medical,woundReceived), [
+                _unit, 
+                1, 
+                _damage, 
+                "unknown", 
+                [[1, _damage / GVAR(structuralDamageToBodyPartCoeff)#1]]
+                ]
+            ] call CBA_fnc_localEvent;
+        };
+        case "vehiclecrash":{
+            TRACE_2("Applying vehicle crash damage",_unit,_damage);
+            private _damageSelectionArray = [ 
+                HITPOINT_INDEX_LARM, 0.1, HITPOINT_INDEX_RARM, 0.1, 
+                HITPOINT_INDEX_LLEG, 1, HITPOINT_INDEX_RLEG, 1
+            ];
+            private _damagedBodyPart = selectRandomWeighted _damageSelectionArray;
+            private _bodyPartDamage = [[_damagedBodyPart, (_damage min PENETRATION_THRESHOLD)  / (GVAR(structuralDamageToBodyPartCoeff) select _damagedBodyPart)]];
+
+            // Significant damage suggests high relative velocity
+            // Momentum transfers to body/head for worse wounding
+            // Higher momentum results in higher chance for head to be hit for more lethality
+            if (_damage > PENETRATION_THRESHOLD) then {
```

Comment:
Vehicle crashes are related to bullet penetration now?

Reply:
AFAIR just suitable constant.